torch/nn/modules/loss.py

```diff
@@ -1254,6 +1257,108 @@ def forward(self, anchor: Tensor, positive: Tensor, negative: Tensor) -> Tensor:
                                      eps=self.eps, swap=self.swap, reduction=self.reduction)
 
 
+class TripletMarginLossWithDistance(_Loss):
+    r"""Creates a criterion that measures the triplet loss given an input
+    tensors :math:`x1`, :math:`x2`, :math:`x3` and a margin with a value greater than :math:`0`.
+    This is used for measuring a relative similarity between samples. A triplet
+    is composed by `a`, `p` and `n` (i.e., `anchor`, `positive examples` and `negative
+    examples` respectively). The shapes of all input tensors should be
+    :math:`(N, D)`.
+
+    The distance swap is described in detail in the paper `Learning shallow
+    convolutional feature descriptors with triplet losses`_ by
+    V. Balntas, E. Riba et al.
+
+    The loss function for each sample in the mini-batch is:
+
+    .. math::
+        L(a, p, n) = \max \{d(a_i, p_i) - d(a_i, n_i) + {\rm margin}, 0\}
+
+
+    where :math:`d(x_i, y_i)` represents the output of `distance_function` on the two
+    inputs.  See also :class:`~torch.nn.TripletMarginLoss`.
+
+    Args:
+        distance_function (callable, optional): A distance function between two Tensors which,
+            if specified, will be used instead of the pairwise distance. If not specified,
+            `nn.PairwiseDistance` will be used.  Default: ``None``
+        is_similarity_function (bool, optional): Whether `distance_function` represents a
+            similarity metrics. Default: ``False``
+        margin (float, optional): Default: :math:`1`.
+        swap (bool, optional): The distance swap is described in detail in the paper
+            `Learning shallow convolutional feature descriptors with triplet losses` by
+            V. Balntas, E. Riba et al. Default: ``False``.
+        reduction (string, optional): Specifies the reduction to apply to the output:
+            ``'none'`` | ``'mean'`` | ``'sum'``. ``'none'``: no reduction will be applied,
+            ``'mean'``: the sum of the output will be divided by the number of
+            elements in the output, ``'sum'``: the output will be summed. Note: :attr:`size_average`
```

Comment:
This comment may need some formatting. If the "size_average" and "reduce" arguments to triplet_margin_loss are deprecated then you can just not include them here.